src/fmureceiver.cpp

```diff
@@ -0,0 +1,307 @@
+    callBackFunctions_.componentEnvironment = fmu_;
+
+    jmStatus_ = fmi2_import_create_dllfmu(fmu_, fmi2_fmu_kind_cs, &callBackFunctions_);
+
+    if (jmStatus_ == jm_status_error)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool
+FMUReceiver::initializeFMU()
+{
+    // TODO: config or use default values?
+    // TODO: alternative solution for start values & verify parameters for setup and initiliazation functions
+    // start values
+    fmi2_string_t instanceName = "Test CS model instance";
+
+    fmi2_string_t fmuLocation = "";
+    fmi2_boolean_t visible = fmi2_false;
+    fmi2_real_t relativeTol = 1e-4;
+
+    tStart_ = 0;
+    tCurrent_ = tStart_;
+    fmi2_boolean_t StopTimeDefined = fmi2_false;
+
+    // Do we need it?
+    fmi2_string_t fmuGUID;
+    fmuGUID = fmi2_import_get_GUID(fmu_);
+
+    jmStatus_ = fmi2_import_instantiate(fmu_, instanceName, fmi2_cosimulation, fmuLocation, visible);
+    fmiStatus_ = fmi2_import_setup_experiment(fmu_, fmi2_true, relativeTol, tStart_, StopTimeDefined, tEnd_);
+    fmiStatus_ = fmi2_import_enter_initialization_mode(fmu_);
+    fmiStatus_ = fmi2_import_exit_initialization_mode(fmu_);
+
+    vr_[FMI_INTEGER_SENSORDATA_IN_BASELO_IDX] = FMI_INTEGER_SENSORDATA_IN_BASELO_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_IN_BASEHI_IDX] = FMI_INTEGER_SENSORDATA_IN_BASEHI_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_IN_SIZE_IDX] = FMI_INTEGER_SENSORDATA_IN_SIZE_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_OUT_BASELO_IDX] = FMI_INTEGER_SENSORDATA_OUT_BASELO_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_OUT_BASEHI_IDX] = FMI_INTEGER_SENSORDATA_OUT_BASEHI_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_OUT_SIZE_IDX] = FMI_INTEGER_SENSORDATA_OUT_SIZE_IDX;
+
+    integerVars_[FMI_INTEGER_SENSORDATA_IN_BASELO_IDX] = 0;
+    integerVars_[FMI_INTEGER_SENSORDATA_IN_BASEHI_IDX] = 0;
+    integerVars_[FMI_INTEGER_SENSORDATA_IN_SIZE_IDX] = 0;
+
+    integerVars_[FMI_INTEGER_SENSORDATA_OUT_BASELO_IDX] = 0;
+    integerVars_[FMI_INTEGER_SENSORDATA_OUT_BASEHI_IDX] = 0;
+    integerVars_[FMI_INTEGER_SENSORDATA_OUT_SIZE_IDX] = 0;
+
+    /* Boolean Variables defined in the OSMPCNetworkProxy.h */
+    //    #define FMI_BOOLEAN_DUMMY_IDX 0
+    //    #define FMI_BOOLEAN_SENDER_IDX 1
+    //    #define FMI_BOOLEAN_RECEIVER_IDX 2
+    fmi2_boolean_t booleanVars_[3];
+    booleanVars_[0] = 0;
+    booleanVars_[1] = 0;
+    booleanVars_[2] = 1;
+    fmi2_status_t status1 = fmi2_import_set_boolean(fmu_, vr_, 3, booleanVars_);
+
+    /* String Variables */
+    //    #define FMI_STRING_ADDRESS_IDX 0
+    //    #define FMI_STRING_PORT_IDX 1
+
+    fmi2_string_t stringVars_[2];
+    stringVars_[0] = ip_.c_str();
+    stringVars_[1] = port_.c_str();
+    fmi2_status_t status2 = fmi2_import_set_string(fmu_, vr_, 2, stringVars_);
+
+    return status1 == fmi2_status_ok && status2 == fmi2_status_ok;
+}
+
+bool
+FMUReceiver::get_fmi_sensor_data_in(osi::SensorData& data)
+{
+    fmiStatus_ = fmi2_import_get_integer(fmu_, vr_, FMI_INTEGER_VARS, integerVars_);
```

Comment:
This also gets the inputs, which does not seem to make sense for the code base here; since inputs are not set anywhere here, they should not be accessed at all, it seems to me.